src/render.c

```diff
@@ -0,0 +1,226 @@
+#include "blokus.h"
+#include "render.h"
+#include <ncurses.h>
+#include <string.h>
+
+const coord_t TILE_CURS[SHAPE_Z + 1] = {
+    {.y = 15, .x = 70},  // SHAPE_O 
+    {.y = 15, .x = 62},  // SHAPE_R 
+    {.y = 15, .x = 55},  // SHAPE_M 
+    {.y = 14, .x = 49},  // SHAPE_A 
+    {.y = 2, .x = 37},   // SHAPE_E 
+    {.y = 7, .x = 37},   // SHAPE_J 
+    {.y = 6, .x = 42},   // SHAPE_S 
+    {.y = 10, .x = 40},  // SHAPE_Q 
+    {.y = 11, .x = 35},  // SHAPE_H 
+    {.y = 11, .x = 69},  // SHAPE_F 
+    {.y = 14, .x = 39},  // SHAPE_I 
+    {.y = 2, .x = 50},   // SHAPE_L 
+    {.y = 2, .x = 68},   // SHAPE_N 
+    {.y = 6, .x = 46},   // SHAPE_P 
+    {.y = 10, .x = 62},  // SHAPE_T 
+    {.y = 11, .x = 48},  // SHAPE_U 
+    {.y = 7, .x = 52},   // SHAPE_V 
+    {.y = 11, .x = 56},  // SHAPE_W 
+    {.y = 6, .x = 68},   // SHAPE_X 
+    {.y = 2, .x = 58},   // SHAPE_Y 
+    {.y = 6, .x = 61},   // SHAPE_Z 
+};
+
+const coord_t ALPHA_CURS[SHAPE_Z + 1] = {
+    {.y = 14, .x = 70},  // SHAPE_O
+    {.y = 14, .x = 63},  // SHAPE_D
+    {.y = 14, .x = 55},  // SHAPE_M
+    {.y = 13, .x = 48},  // SHAPE_A
+    {.y = 0, .x = 38},   // SHAPE_E
+    {.y = 4, .x = 36},   // SHAPE_J
+    {.y = 4, .x = 41},   // SHAPE_S
+    {.y = 9, .x = 41},   // SHAPE_Q
+    {.y = 9, .x = 36},   // SHAPE_H
+    {.y = 9, .x = 69},   // SHAPE_F
+    {.y = 13, .x = 39},  // SHAPE_I
+    {.y = 0, .x = 47},   // SHAPE_L
+    {.y = 0, .x = 67},   // SHAPE_N
+    {.y = 4, .x = 47},   // SHAPE_P
+    {.y = 9, .x = 62},   // SHAPE_T
+    {.y = 9, .x = 48},   // SHAPE_U
+    {.y = 4, .x = 54},   // SHAPE_V
+    {.y = 9, .x = 56},   // SHAPE_W
+    {.y = 4, .x = 68},   // SHAPE_X
+    {.y = 0, .x = 57},   // SHAPE_Y
+    {.y = 4, .x = 61},   // SHAPE_Z
+};
+
+const shape_t tile_relation[4][6] = {
+    {SHAPE_E, SHAPE_E, SHAPE_L, SHAPE_Y, SHAPE_Y, SHAPE_N},
+    {SHAPE_J, SHAPE_S, SHAPE_P, SHAPE_V, SHAPE_Z, SHAPE_X},
+    {SHAPE_H, SHAPE_Q, SHAPE_U, SHAPE_W, SHAPE_T, SHAPE_F},
+    {SHAPE_I, SHAPE_I, SHAPE_A, SHAPE_R, SHAPE_D, SHAPE_M}
+};
+
+int init_all_colors() {
+    start_color();
+    init_pair(RED_PAIR, COLOR_RED, COLOR_BLACK);
+    init_pair(GREEN_PAIR, COLOR_GREEN, COLOR_BLACK);
+    init_pair(YELLOW_PAIR, COLOR_YELLOW, COLOR_BLACK);
+    init_pair(CYAN_PAIR, COLOR_CYAN, COLOR_BLACK);
+    return 0;
+}
+
+int render_board(gcb_t *gcb)
+{
+    mvprintw(11, 11, "+");
+    mvprintw(6, 21, "+");
+    mvprintw(1, 0, "   0 1 2 3 4 5 6 7 8 9 a b c d  │\n");
+    for (int y = 0; y < N_ROW; ++y) {
+        printw(" %x ", y);
+        for (int x = 0; x < N_COL; ++x) {
+            if (gcb->map[y][x] != -1) {
+                printw("%s ", (gcb->map[y][x])? "□" : "■"); 
+            } else {
+                printw("%s ", (((y == 4 && x == 9) || (y == 9 && x == 4)))? "+" : "·"); 
```

Comment:
Let's wrap this line since it exceeds 80 characters.
```suggestion
                printw("%s ",
                       (((y == 4 && x == 9) || (y == 9 && x == 4)))? "+" : "·"); 
```